src/decorator_trait.jl

```diff
@@ -330,11 +361,21 @@ function injectivity_radius(
     return injectivity_radius(get_embedding(M, p), p, m)
 end
 
-# Introduce Deco Trait | automatic foward | fallback
-@trait_function inner(M::AbstractDecoratorManifold, p, X, Y)
-# Isometric Embedded submanifold
-function inner(
-    ::TraitList{IsIsometricEmbeddedManifold},
+function inner(M::AbstractDecoratorManifold, p, X, Y)
+    return _inner_embedding(get_embedding_type(M), M, p, X, Y)
+end
```

Comment:
@kellertuer This follows the new pattern for forwarding inner, although it currently conflicts with `IsExplicitDecorator` (specifically, the `FallbackManifold` tests). Are we replacing `IsExplicitDecorator` with explicit calls or something else?